### ISSUE TITLE
Assertion error during is_zero call

### ISSUE THREAD

**Issue by proy87** [sympy/sympy#26081]
`n = Symbol('n', integer=True)`
`((-3*(-1)**n + 4*n**2 + 8*n + 3)/(2*(-1)**n)).is_zero` raises AssertionError.

**Comment by oscarbenjamin** [comment]
I can't reproduce this with current master. Which version are you using?

**Comment by proy87** [comment]
I use 1.9. Possibly in later versions the bug is fixed.

**Comment by oscarbenjamin** [comment]
I can reproduce with 1.9:
```python
In [1]: n = Symbol('n', integer=True)

In [2]: ((-3*(-1)**n + 4*n**2 + 8*n + 3)/(2*(-1)**n)).is_zero
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
File /space/enojb/sympy/sympy.git/sympy/core/assumptions.py:462, in make_property.<locals>.getit(self)
    461 try:
--> 462     return self._assumptions[fact]
    463 except KeyError:

KeyError: 'zero'

During handling of the above exception, another exception occurred:

KeyError                                  Traceback (most recent call last)
File /space/enojb/sympy/sympy.git/sympy/core/assumptions.py:462, in make_property.<locals>.getit(self)
    461 try:
--> 462     return self._assumptions[fact]
    463 except KeyError:

KeyError: 'odd'

During handling of the above exception, another exception occurred:

AssertionError                            Traceback (most recent call last)
Cell In[2], line 1
----> 1 ((-3*(-1)**n + 4*n**2 + 8*n + 3)/(2*(-1)**n)).is_zero

File /space/enojb/sympy/sympy.git/sympy/core/assumptions.py:466, in make_property.<locals>.getit(self)
    464 if self._assumptions is self.default_assumptions:
    465     self._assumptions = self.default_assumptions.copy()
--> 466 return _ask(fact, self)

File /space/enojb/sympy/sympy.git/sympy/core/assumptions.py:521, in _ask(fact, obj)
    519     continue
    520 if pk in handler_map:
--> 521     _ask(pk, obj)
    523     # we might have found the value of fact
    524     ret_val = assumptions.get(fact)

File /space/enojb/sympy/sympy.git/sympy/core/assumptions.py:509, in _ask(fact, obj)
    507     pass
    508 else:
--> 509     a = evaluate(obj)
    510     if a is not None:
    511         assumptions.deduce_all_facts(((fact, a),))

File /space/enojb/sympy/sympy.git/sympy/core/mul.py:1613, in Mul._eval_is_even(self)
   1610 is_integer = self.is_integer
   1612 if is_integer:
-> 1613     return fuzzy_not(self.is_odd)
   1615 n, d = fraction(self)
   1616 if n.is_Integer and n.is_even:
   1617     # if minimal power of 2 in den vs num is not
   1618     # negative then this is not an integer and
   1619     # can't be even

File /space/enojb/sympy/sympy.git/sympy/core/assumptions.py:466, in make_property.<locals>.getit(self)
    464 if self._assumptions is self.default_assumptions:
    465     self._assumptions = self.default_assumptions.copy()
--> 466 return _ask(fact, self)

File /space/enojb/sympy/sympy.git/sympy/core/assumptions.py:509, in _ask(fact, obj)
    507     pass
    508 else:
--> 509     a = evaluate(obj)
    510     if a is not None:
    511         assumptions.deduce_all_facts(((fact, a),))

File /space/enojb/sympy/sympy.git/sympy/core/mul.py:1595, in Mul._eval_is_odd(self)
   1593 if abs(t) is S.One:
   1594     continue
-> 1595 assert t.is_integer
   1596 if t.is_even:
   1597     return False

AssertionError: 
```

This was fixed in 1.11 though due to 95a13732c1c2d2b9fe8de01075c4d5eef37492c1 from gh-22557 which largely rewrote the relevant section of code.

I think we can close this. Thanks for reporting.
